src/frontend/hooks/useDraftObservation.ts

```diff
@@ -75,6 +76,33 @@ export const useDraftObservation = (): UseDraftObservation => {
   );
   const [savingStatus, setSavingStatus] = useState<Status>();
 
+  function deletePhoto(uri: string) {
+    const newPhotosArray = draft.photos.filter(
+      photo => "originalUri" in photo && photo.originalUri !== uri
+    );
+
+    const newPhotoPromiseArray = draft.photoPromises.map(async photo => {
+      const resolvedPhoto = await photo;
+      if (resolvedPhoto.originalUri === uri) {
+        const deletedPhoto: Promise<DraftPhoto> = new Promise((res, rej) => {
+          resolvedPhoto.deleted = true;
+          res(resolvedPhoto);
+        });
+        const cancelPhoto = deletedPhoto as CancellablePromise<DraftPhoto>;
+        cancelPhoto.signal = { didCancel: true };
```

Comment:
Not really following functional programming here, seemed a little clunky to create so 2 new objects